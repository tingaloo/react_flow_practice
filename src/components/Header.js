import React from 'react';
import PropTypes from 'prop-types';
import { Button, Container, Divider, Grid, Image, Menu, Segment, Header } from 'semantic-ui-react';
import { Link } from 'react-router-dom';

import type {User} from '../types/user';

// using flow
export type Props = {
  user: User
}

const HeaderMain = ({ user, loading }: Props) => (

  <div>
  <Header size='huge'> React Boilerplate </Header>
    {loading && !user.username && "Loading"}
    {user.username ? "Welcome "+user.username : "Login will ya?" }

  </div>

)

export default HeaderMain
