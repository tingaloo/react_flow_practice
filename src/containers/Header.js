// @flow

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actionCreators from '../actions';
import Header from '../components/Header';

type Props = {
  // state: {}
}

const mapStateToProps = (state) => {
  return {
    user: state.user,
    loading: state.loginLoading,
    error: state.loginError,
  };
};

const mapDispatchToProps = dispatch => bindActionCreators(actionCreators, dispatch);

const HeaderContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Header);

export default HeaderContainer;
