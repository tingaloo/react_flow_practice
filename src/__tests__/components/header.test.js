import React from 'react';
import ReactDOM from 'react-dom';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../../components/Header';

const user = {
  username: 'fda',
  password: 'password',
};

it('renders without crashing with vanilla react', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Header user={user} />
    , div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// using enzyme
it('renders without crashing with enzyme', () => {
  let userWrongType = [];
  const wrapper = shallow(<Header user={user}/>);
  expect(wrapper.contains('h1'));
});

//using jest
test('renders without crashing with jest snapshots', () => {
  const john = {
    username: 'john',
    password: 'wick',
  };
  const wrapper = shallow(
    <Header user={john}/>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
