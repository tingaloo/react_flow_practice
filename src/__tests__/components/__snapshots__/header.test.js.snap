// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing with jest snapshots 1`] = `
<div>
  <h1>
     Header 
  </h1>
  Welcome john
</div>
`;
